### DISCUSSION TITLE
Builds with macOS SDK are failing

### DISCUSSION THREAD

**Discussion by Sn0wfreezeDev** [zephyrproject-rtos/zephyr#38573] [Q&A]
**Describe the bug**

When building anything on macOS I receive an error. 

```
west build -p auto -b nrf52dk_nrf52832 samples/basic/blinky
```

```c
/zephyrproject/zephyr/arch/arm/include/aarch32/cortex_m/stack.h:29:8: error: argument to 'section' attribute is not valid for this target: mach-o section specifier requires a segment whose length is between 1 and 16 characters
extern K_KERNEL_STACK_ARRAY_DEFINE(z_interrupt_stacks, CONFIG_MP_NUM_CPUS,
       ^
```

**To Reproduce**

I downloaded the SDK and installed the toolchain for macOS. 
I strictly followed the tutorial on the [Zephyr Website](https://docs.zephyrproject.org/latest/getting_started/index.html), which seems outdated for macOS, as it says it's not supported. 

So I followed the [Nordic getting started guide](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/0.4.0/zephyr/getting_started/index.html) for setting up the toolchain. 

I set the environment to `ZEPHYR_TOOLCHAIN_VARIANT=host`

**Expected behavior**
The build should complete 

**Impact**
This is a showstopper since the Toolchain does not run 

**Logs and console output**
Can be found above. 

**Environment (please complete the following information):**
 - OS macOS 
 - Toolchain Zephyr SDK
 - Commit cd5357ca56c2629b957a3a79a3dd71fe528e2626

**Comment by carlescufi** [comment]
>I strictly followed the tutorial on the Zephyr Website, which seems outdated for macOS, as it says it's not supported.

That doesn't sound right. The current docs clearly state that macOS is supported:
https://docs.zephyrproject.org/latest/getting_started/index.html

>I set the environment to ZEPHYR_TOOLCHAIN_VARIANT=host

I am not quite sure this is right. @tejlmand ?

**Reply by Sn0wfreezeDev** [reply]
In a later section the websites states 

> Follow the instructions in Set Up a Toolchain. Note that the Zephyr SDK is not available on macOS.
> Do not forget to set the required environment variables (ZEPHYR_TOOLCHAIN_VARIANT and toolchain specific ones).

But the newest SDK is available on macOS and a very similar setup compared to Linux is necessary.

**Reply by stephanosio** [reply]
> But the newest SDK is available on macOS and a very similar setup compared to Linux is necessary.

As noted earlier, the macOS SDK is currently experimental and not really "available" for production use; though, we could probably make this more clear.

cc @galak

**Comment by stephanosio** [comment]
There are quite a few problems here:

> I set the environment to `ZEPHYR_TOOLCHAIN_VARIANT=host`
> ...
> Toolchain Zephyr SDK

If you set `ZEPHYR_TOOLCHAIN_VARIANT` to `host`, you are not using the Zephyr SDK.

> `west build -p auto -b nrf52dk_nrf52832 samples/basic/blinky`

You cannot use the host toolchain to build ARM boards.

p.s. you should be able to build for the `nrf52dk_nrf52832` on macOS if you use the macOS Zephyr SDK (with `ZEPHYR_TOOLCHAIN_VARIANT=zephyr`); though, we have not tested it, and it is still very much experimental.

**Reply by Sn0wfreezeDev** [reply]
Thank you for the clarification here. 

The environment variable `ZEPHYR_TOOLCHAIN_VARIANT` was a bit unclear to me. I understand that host should not work for ARM boards, but I did not find other options to set on the [website](https://docs.zephyrproject.org/latest/guides/beyond-GSG.html?highlight=zephyr_toolchain_variant). It just states: 
```
You configure the Zephyr build system to use a specific toolchain by setting environment variables such as ZEPHYR_TOOLCHAIN_VARIANT to a supported value, along with additional variable(s) specific to the toolchain variant.
```

Setting the toolchain to *zephyr*, cleaning the build folder and rebuilding worked! 

I also installed the Zephyr SDK on Linux and it does not require to set the `ZEPHYR_TOOLCHAIN_VARIANT`. On macOS starting a build without a specified toolchain would result this error: 
```
 ZEPHYR_TOOLCHAIN_VARIANT not specified and no Zephyr SDK is installed.

  Please set ZEPHYR_TOOLCHAIN_VARIANT to the toolchain to use or install the
  Zephyr SDK.
```
This is after the Zephyr SDK was installed

**Reply by stephanosio** [reply]
> The environment variable `ZEPHYR_TOOLCHAIN_VARIANT` was a bit unclear to me. I understand that host should not work for ARM boards, but I did not find other options to set on the [website](https://docs.zephyrproject.org/latest/guides/beyond-GSG.html?highlight=zephyr_toolchain_variant).

Officially, Zephyr SDK is currently not supported on macOS (it is experimental) right now.

There is work being done to improve multi-platform support (see #37255) -- once that is done, we will update the "Getting Started Guide" with more beginner-friendly and consistent set-up procedures.
